crypto/asn1/a_int.c

```diff
@@ -251,7 +251,7 @@ static int asn1_get_int64(int64_t *pr, const unsigned char *b, size_t blen,
             ASN1err(ASN1_F_ASN1_GET_INT64, ASN1_R_TOO_SMALL);
             return 0;
         }
-        *pr = (int64_t)(0 - r);
+        *pr = -(int64_t)r;
```

Comment:
No, VS takes issue with unary minus specifically with *unsigned* values. It's not a problem to apply unary minus with *signed* values. And explicit cast does yield signed value, so VS is OK with it. [I did check by the way.]